### DISCUSSION TITLE
[security] Is GITHUB_TOKEN different for each Action run?

### DISCUSSION THREAD

**Discussion by llucax** [community/community#26559] [Actions]
<p>Hi, I’m having this question because of the <a href="https://about.codecov.io/security-update/" rel="noopener nofollow ugc">Codecov uploader security breach</a>, which has exposed all environment variables (including <code>GITHUB_TOKEN</code>. Since I have no control over this particular environment variable, I wonder if it is generated for each run (so I shouldn’t worry about it being exposed) or if it persist across runs (so I should worry a lot).</p>
<p>If it’s not generated on each run, is there any way to regenerate it?</p>
<p>Thanks!</p>

**Comment by ylemkimon** [comment, marked as answer]
<p>Yes, see <a href="https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret:" class="inline-onebox" rel="noopener nofollow ugc">Authentication in a workflow - GitHub Docs</a></p>
<blockquote>
<p>Before each job begins, GitHub fetches an installation access token for the job. The token expires when the job is finished.</p>
</blockquote>

**Comment by llucax** [comment]
<p>Hi <a class="mention" href="/ylemkimon">@ylemkimon</a>,</p>
<p>I read that article but it wasn’t clear to me that the token was rotated because I’m not familiar with the internals of how GitHub Apps works.</p>
<p>Are you sure the documentation there really implies that the token is different for each run/GitHub App installation?</p>
<p>Thanks!</p>

**Comment by ylemkimon** [comment]
<p>Yes. If in doubt, you can always test it yourself, e.g., <code>echo -n "${{ github.token }}" | sha256sum</code> (note it’d be <strong>possible to infer</strong> the token from the hash, as the token has a fixed length and characters, so do not test this in a public or real repository). This will give a different hash every time the workflow is run.</p>

**Comment by llucax** [comment]
<p>Thanks again for the answer. I thought of doing something like that to get some empirical proof, but then only because an experiment showed something doesn’t mean it is guaranteed to change, so I have more peace of mind if there is a guarantee that different tokens will be created for different runs.</p>
<p>In any case, I consider this question answered, so thanks again!</p>
